File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    static {
        for (int i = 0; i < 500; i++) {
            moveZeroes(new int[] {});
        }
    }

    public static void moveZeroes(int[] nums) {
        int z = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[z];
                nums[z] = temp;
                z++;
            }
        }
    }
}

File: LeetCode/Easy/0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int start = 0, end = 1;

        while (end < nums.size()) {
            if (nums[start] == 0) {
                if (nums[end] == 0) {
                    end++;
                    continue;
                }

                swap(nums[end], nums[start]);
            }

            start++;
            end++;
        }
    }
};